pkg/internal/flags.go

```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+// Used by tests to selectively disable experimental JSON unmarshaler
+var UseExperimentalJSONUnmarshaling bool = true
```

Comment:
if we're going to enable this by default, consider naming this something like `UseOptimizedJSONUnmarshaling` ... enabling an "experiment" by default with no way for consumers to opt out is anxiety-inducing (despite the name of the dependency)